trade-finance-manager-ui/server/controllers/case/cancellation/check-details.controller.post.test.ts

```diff
@@ -48,41 +55,128 @@ describe('postBankRequestDate', () => {
-
-    const { req, res } = createMocks<PostDealCancellationDetailsRequest>({
-      params: { _id: dealId },
-      session: aRequestSession(),
+  describe(`when the deal type is ${DEAL_SUBMISSION_TYPE.MIA}`, () => {
+    beforeEach(() => {
+      jest.mocked(api.getDeal).mockResolvedValue({ dealSnapshot: { details: { ukefDealId }, submissionType: DEAL_SUBMISSION_TYPE.MIA } });
     });
 
-    // Act
-    await postDealCancellationDetails(req, res);
+    it('redirects to deal summary page', async () => {
+      // Arrange
+      const { req, res } = createMocks<PostDealCancellationDetailsRequest>({
+        params: { _id: dealId },
+        session: aRequestSession(),
+        body: { cancellation: { reason, bankRequestDate, effectiveFrom } },
+      });
 
-    // Assert
-    expect(res._getRedirectUrl()).toEqual(`/case/${dealId}/deal`);
-  });
+      // Act
+      await postDealCancellationDetails(req, res);
 
-  describe.each([DEAL_SUBMISSION_TYPE.AIN, DEAL_SUBMISSION_TYPE.MIN])('when the deal type is %s', (validDealType) => {
-    beforeEach(() => {
-      jest.mocked(api.getDeal).mockResolvedValue({ dealSnapshot: { details: { ukefDealId }, submissionType: validDealType } });
+      // Assert
+      expect(res._getRedirectUrl()).toEqual(`/case/${dealId}/deal`);
     });
 
-    // TODO: DTFS2-7298 - test deal cancellation endpoints are called with correct payload
-
-    it('redirects to the effective from date page', async () => {
+    it('does not submit a deal cancellation', async () => {
       // Arrange
       const { req, res } = createMocks<PostDealCancellationDetailsRequest>({
         params: { _id: dealId },
         session: aRequestSession(),
+        body: { cancellation: { reason, bankRequestDate, effectiveFrom } },
       });
 
       // Act
       await postDealCancellationDetails(req, res);
 
       // Assert
-      expect(res._getRedirectUrl()).toEqual(`/case/${dealId}/deal`);
+      expect(api.submitDealCancellation).toHaveBeenCalledTimes(0);
+    });
+  });
+
+  describe.each([DEAL_SUBMISSION_TYPE.AIN, DEAL_SUBMISSION_TYPE.MIN])('when the deal type is %s', (validDealType) => {
+    beforeEach(() => {
+      jest.mocked(api.getDeal).mockResolvedValue({ dealSnapshot: { details: { ukefDealId }, submissionType: validDealType } });
+    });
+
+    const invalidTestCases = [
+      {
+        description: 'when the reason is undefined',
+        cancellation: { bankRequestDate, effectiveFrom },
+      },
+      {
+        description: 'when the bankRequestDate is undefined',
+        cancellation: { reason, effectiveFrom },
+      },
+      {
+        description: 'when the effectiveFrom is undefined',
+        cancellation: { reason, bankRequestDate },
+      },
+    ];
+
+    describe.each(invalidTestCases)('$description', ({ cancellation }) => {
+      it('does not submit the deal cancellation', async () => {
```

Comment:
🤩 Where is this logic handled in the ui layer? 🕵️

Reply:
`    const cancellationIncomplete = reason === undefined || !bankRequestDate || !effectiveFrom;` deals with this - Its obviously not a perfect check but it keeps Typescript happy & it shouldnt ever be needed